#1387: fix error after switching on saved view

mmezhensky opened the pull request:
# Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

This PR fixed issue [1387](https://github.com/magento/adobe-stock-integration/issues/1387).

### Fixed Issues
<!---
    If relevant, please provide a list of fixed issues in the format magento/adobe-stock-integration#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. Fixes magento/adobe-stock-integration#1387: "Uncaught TypeError: Cannot read property 'complete' of undefined" appears in dev console if save Previewed image as a new View and open this View on another page

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Go to Content - Media Gallery
2. Click Search Adobe Stock
3. Go to the first page from the grid, and select any image to open its Preview
4. Click on Default View (bookmarks view)
5. Click Save View As...,
6. Click Save all Changes arrow_right
7. Switch to the Default View
8. Go to the second page in Adobe Stock grid
9. Switch to the previously created New View

**>> No error message in dev console**

Commit:
#1387: fix error after switching on saved view

gabrieldagama reviewed (changes requested):
Hi @mmezhensky, thanks for the pull request!

I've updated the issue with a comment, but basically we have another bug when performing the same steps described on the issue. The bug is the image that is displayed after switching bookmarks is the wrong one. Follow below some suggestions on these issues:

So, related to the console error, I believe we should handle that on magento2 repository, and on the `app/code/Magento/Ui/view/base/web/js/grid/columns/image-preview.js` component, to solve the issue we can extract the part that is causing the error to an external function and handle it there. The problem seems to be related to the image not being there when the code gets executed, we could do something like this to prevent the error from happening:

```
        updateUserInterface: function () {
            var img = $(this.previewImageSelector + ' img');

            if (!img.get(0)) {
                setTimeout(function () {
                    this.updateUserInterface();
                }.bind(this), 100);
            } else {
                if (img.get(0).complete) {
                    this.updateHeight();
                    this.scrollToPreview();
                } else {
                    img.load(function () {
                        this.updateHeight();
                        this.scrollToPreview();
                    }.bind(this));
                }
            }
        },
```

And about the wrong image being displayed at the image-preview, we probably would have to listen to the data provider and update the image-preview component, that will prevent this issue from happening with bookmarks and other similar scenarios.

Let me know if you have any questions or want to discuss the approaches!

sivaschenko commented:
Hi @mmezhensky , would you like to continue progress on this pull request? Or do you need any help?

Commit:
#912 - fix wrong image being displayed at the image-preview

mmezhensky commented:
Hello @gabrieldagama @sivaschenko 
Sorry, but I didn't have time for this earlier.
Тow I'm back to that, and I created a new commit where I changed fix for the console error and fixed wrong image being displayed at the image-preview.

mmezhensky commented:
@magento run all tests

sivaschenko committed:
Merge branch '2.0-develop' into issue-1387

sivaschenko commented on AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image-preview.js:
```diff
@@ -48,10 +48,33 @@ define([
             listens: {
-                '${ $.sortByComponentName }:applied': 'hide'
+                '${ $.sortByComponentName }:applied': 'hide',
+                'adobe_stock_images_listing.adobe_stock_images_listing.listing_top.bookmarks:activeIndex': 'onActiveIndexChange'
```
Please add the bookmarksProvider to the defaults to be able to configure it for adobe stock listing and standalone adobe stock listing separately:
 - `AdobeStockImageAdminUi/view/adminhtml/ui_component/standalone_adobe_stock_images_listing.xml`
 - `AdobeStockImageAdminUi/view/adminhtml/ui_component/adobe_stock_images_listing.xml`

sivaschenko reviewed (changes requested):
Thanks for updates @mmezhensky ! I have added a comment on the implementation, however, this PR does not fix the issue for me. Can you please recheck it

sivaschenko commented:
Closed due to inactivity

ghost commented:
Hi @mmezhensky, thank you for your contribution! </br>Please, complete [Contribution Survey](https://opensource.magento.com/survey/contribution/magento+adobe-stock-integration+1430), it will take less than a minute. </br>Your feedback will help us to improve contribution process.
